### PR TITLE
Support for advertising identifier and adId consent collection

### PR DESCRIPTION
**Pull request by emdobrin**
## Description

Update for ad ID feature, related test cases, and update to Android test app to use new ad ID features with real-world GAID implementation example.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/app/build.gradle [hunk 1]
```diff
@@ -57,8 +57,14 @@ dependencies {
     implementation project(':edgeidentity')
     implementation 'com.adobe.marketing.mobile:core:1+'
     implementation 'com.adobe.marketing.mobile:identity:1+'
-    implementation('com.adobe.marketing.mobile:edge:1+') {
+    implementation 'com.adobe.marketing.mobile:edgeconsent:1.+'
+    implementation('com.adobe.marketing.mobile:edge:1.3.2-SNAPSHOT') {
```
Comment (cacheung) [inline code/app/build.gradle:61]: We will keep Edge as pointing to Staging before merge to EdgeIdentity Staging?

Comment (emdobrin) [reply code/app/build.gradle:61]: We should update this as soon as Edge 1.3.2 is released, we can make this change in dev branch before merging to staging.

**Review by cacheung — APPROVED** [review]

File: code/app/src/main/java/com/adobe/marketing/edge/identity/app/EdgeIdentityApplication.kt [hunk 2]
```diff
@@ -16,11 +16,12 @@ import com.adobe.marketing.mobile.Assurance
 import com.adobe.marketing.mobile.Edge
 import com.adobe.marketing.mobile.LoggingMode
 import com.adobe.marketing.mobile.MobileCore
+import com.adobe.marketing.mobile.edge.consent.Consent
 import com.adobe.marketing.mobile.edge.identity.Identity
 
 class EdgeIdentityApplication : Application() {
-    // Add your Launch Environment ID to configure the SDK from your Launch property
-    private var LAUNCH_ENVIRONMENT_ID: String = ""
+    // Add your app environment id to configure the SDK from your Data Collection mobile property
```
Comment (cacheung) [inline code/app/src/main/java/com/adobe/marketing/edge/identity/app/EdgeIdentityApplication.kt:23]: @timkimadobe we can update this to the same as iOS when you update the test app in the next PR.

// TODO: Set up the preferred Environment File ID from your mobile property configured in Data Collection UI
   
private var ENVIRONMENT_FILE_ID: String = "yourAppId"

MobileCore.configureWithAppID(ENVIRONMENT_FILE_ID)

File: README.md [hunk 3]
```diff
@@ -20,11 +20,20 @@ implementation 'com.adobe.marketing.mobile:core:1.+'
...
+
+The test app needs to be configured with the following edge extensions before it can be used:
+- Mobile Core (installed by default)
+- [Edge](https://aep-sdks.gitbook.io/docs/foundation-extensions/experience-platform-extension)
+- [Edge Identity](https://aep-sdks.gitbook.io/docs/foundation-extensions/identity-for-edge-network)
 
 **Run demo application**
 
-Once you opened the project in Android Studio (see above), select the `app` runnable and your favorite simulator and run the program.
+1. In the test app, set your APP_ENVIRONMENT_ID in EdgeIdentityApplication.kt.
```
Comment (kevinlind) [inline README.md:34]: Should be `ENVIRONMENT_FILE_ID`.

Comment (timkimadobe) [reply README.md]: Updated!

File: code/app/src/main/java/com/adobe/marketing/edge/identity/app/EdgeIdentityApplication.kt (hunk 2 above)
Comment (timkimadobe) [reply code/app/src/main/java/com/adobe/marketing/edge/identity/app/EdgeIdentityApplication.kt]: Updated to the new naming style!

**Pull request merged**